### ISSUE TITLE
Asepsis 1.5 not working on OS X 10.10.1

### ISSUE THREAD

**Issue by revintec** [binaryage/asepsis#23]
I don't know what kind of information I should provide
I just wanna know if anyone has the same problem
or anyone tested that asepsis is working on 10.10.1?

The problem is:
After a successful installation of Asepsis 1.5(which is the latest)
I've removed all .DS_Store and immediately rebooted
after some day, still see a lot of .DS_Store file everywhere
(I'm a developer, and I turned on ShowHiddenFiles in every app)
and there're also a lot _DS_Store in asepsis dscage

asepsisctl diagnose says it's all OK

installed XtraFinder, don't know if that will cause a problem
XtraFinder should not interfere with finder in that way
it says it was designed to add extra functionality to Finder
(like "Cut file", "New file", "Refresh" in context menu and etc.)
no mention of anything related to DS_Store in the app

My OS X 10.10.1 may have some problem
when I've noticed dtrace do not work properly
especially execsnoop is not working
but otherwise it seems working fine
and that shouldn't interfere with asepsis

**Comment by chodorowicz** [comment]
I have same situation. Asepsis diagnose says all is ok, but DS_Store files are still being everywhere around the drives.

**Comment by revintec** [comment]
and it causes problems
when move ds_store on my desktop,
my icon arrangement got destroyed.
that's why I uninstalled asepsis.

it's working partially,
some ds_store got moved while others are not,
I think we need to hook more functions

**Comment by darwin** [comment]
Is this still an issue under 10.10.3+? Asepsis 1.5 seems to be working fine for me.

**Comment by chodorowicz** [comment]
It seems to be working for me now.

**Comment by darwin** [comment]
Good! Closing. Feel free to reopen if needed.

**Comment by revintec** [comment]
sorry for the late comment, did some investigation.

I'm now using 10.10.3, and 1.5(will try the latest 1.5.1 later).
Asepsis is carefully designed(which only monitors a private framework),
so if I extract a .zip or .rar that contains .DS_Store files,
these .DS_Store files will appear in Finder, and asepsis won't remove them.
(this is literally perfect, no, I'm not be sarcasm,
this means asepsis won't interfere with file ops)
(asepsis... but these files are germs, tooㄟ( ▔, ▔ )ㄏ )
That's the reason(but not the only reason) why I'm still having some .DS_Store files.

Further investigation shows that some Apple apps still managed to create .DS_Store files,
yes, I'm looking at you, Xcode! can't reproduce the result reliably,
but sometimes, when using Open/Save dialogs,
the .DS_Store s got created(first, they were not there and then they pops up)

**Comment by darwin** [comment]
Asepsis is white-listed only for Finder and mdworker right now:
https://github.com/binaryage/asepsis/blob/master/wrapper/init.c#L21-L25

Maybe we should add Xcode (or the process which is causing them) there as well.

**Comment by revintec** [comment]
Yes, I'm aware of that.

I'm still investigating, try to make sure what is really responsible for creating all these .DS_Store files.
I can't reproduce the result reliably, so my guess is that Xcode may be not responsible,
maybe the Open/Save dialog framework?

The thing is that, I have a folder A to store some utils I wrote.
I access this folder through:
Finder: should not create .DS_Store since I've installed Asepsis
Xcode(save compiled code, open configuration files, etc.) -- suspect
iTerm, git, Sublime... -- also suspect, but unlikely...
